0002/README.md

```diff
@@ -0,0 +1,167 @@
+---
+id: 0002
+title: Set the mood around releasing code
+status: ideation
+authors: Gianluca Arbezzano <gianarb92@gmail.com>
+---
+
+## Summary
+
+This proposal sets the expectation about what it means to release code part of
+the Tinkerbell organisation.
+
+## Goals and not Goals
+
+Goals are:
+
+1. What is a release
+2. Set a few rules that all the projects should follow
+3. Quality assertion.
+
+Not-Goals:
+
+1. How each project should be released
+2. Define who own releases
+3. Distribution via `apt`, `apk` and so on.
+
+## Content
+
+A release is a tested and stable point in time snapshot from a codebase.
+
+We follow the [Semantic Versioning (semver) 2.0.0](https://semver.org).
+Summarized as follows:
+
+Given a version number MAJOR.MINOR.PATCH, increment the:
+
+* MAJOR version when you make incompatible API changes,
+* MINOR version when you add functionality in a backwards compatible manner, and
+* PATCH version when you make backwards compatible bug fixes.
+
+Additional labels for pre-release and build metadata are available as extensions
+to the MAJOR.MINOR.PATCH format.
+
+### Schedule
+
+Every subproject should be able to follow its own cadence in terms of how often
+or regularly a new release should be cut, as long as it is documented and
+shared with the community.
+
+A new `MINOR` version for Tinkerbell should be released following a proper
+schedule, right now I think every two months sounds reasonable looking at the
+code written and merged in the last couple of weeks. In this way the
+subprojects will have an intrinsic release target if they want to get their new
+code included in the Tinkerbell release.
+
+`PATCH` release for Tinkerbell can be done as required by the subprojects
+bug fixes.
+
+`MAJOR` release right now are not even evaluated because the project is too young.
+
+### GitHub release and release website
+
+Git supports tag and GitHub has the concept of a release based on a particular
+tag.
+
+The artifact generated as part of the release will be published as part of the
+release website, using as a guidance how [HashiCorp
+Release](https://releases.hashicorp.com/) works.
+
+I like the idea to use a release website vs having to push them as manifest in
+GitHub release because it gives a single place where the community can
+programmatically lookup what they are looking for, without having to look for
+repositories, versions and so on.
```

Comment:
We could leverage something like goreleaser (at least for go projects) to automate the release process using github releases, and we can leverage the `latest` release link for the common use cases.

That might make a decent solution, at least for individual subprojects.

Reply:
Yes, the problem with goreleaser is that it does not work great when it comes to multi arch support for Docker containers

Reply:
Some excellent resources here:
* https://github.com/goreleaser/goreleaser/issues/530
* https://www.youtube.com/watch?v=xdehEUG1RFY 😄

Reply:
Am I wrong to think that maintaining an independent release website is going to be a burden on project maintainers?

If this is just object storage (url+key) as you suggest below as an option for distribution we may not be able to get it for free, but I think maintenance will be less cumbersome than bintray.

> using as a guidance how [HashiCorp Release](https://releases.hashicorp.com/) works

The Hashicorp example is interesting because they have moved away from releases.hashicorp.com for Terraform providers (now contained in a registry).

> I like the idea to use a release website vs having to push them as manifest in GitHub release because it gives a single place where the community can programmatically lookup what they are looking for

The GitHub releases can be accessed through the [GitHub API](https://developer.github.com/v3/repos/releases/#list-releases (`GET)
) very easily:
```sh
$ curl https://api.github.com/repos/tinkerbell/tink/releases
[

]
```

> without having to look for repositories, versions and so on

I think this can be solved through the GitHub `latest` keyword being used in URLs in-place of a version, and by distributing our binaries through other systems (docker, debian, fedora, etc) where users won't need to worry about versions (maintainers will).

Reply:
I'm wondering if using Github releases just saves us time and effort - it's a solved problem with no overhead.

Reply:
I recall the interest in nightly builds that are not based on tags or releases. I think this requirement is perhaps causing us more grief than it provides in value.

Reply:
I'm wondering if we can simplify the organization's requirements, for now, and prevent the need for https://github.com/tinkerbell/proposals/pull/5/files#diff-5ae2d202d9cad195081db97be153eaa9R101-R106.

A maintenance team would certainly be needed to coordinate with interested vendors to prepare, build, certify, and deploy tink in their environments.  Do we have to figure this out now? Can this infrastructure maintenance team (and the additional infrastructure requirements) be the target of a future proposal?